File: Recien traducido/Grafos/MaximumBipartiteMatching.cpp
#include <iostream> 
#include <string.h> 
using namespace std;
#define M 6 
#define N 6 

bool bpm(bool bpGraph[M][N], int u,
        bool seen[], int matchR[]) {

    for (int v = 0; v < N; v++) {

        if (bpGraph[u][v] && !seen[v]) {

            seen[v] = true;

            if (matchR[v] < 0 || bpm(bpGraph, matchR[v],
                    seen, matchR)) {
                matchR[v] = u;
                return true;
            }
        }
    }
    return false;
}

int maxBPM(bool bpGraph[M][N]) {

    int matchR[N];
    memset(matchR, -1, sizeof (matchR));

    int result = 0;
    for (int u = 0; u < M; u++) {
        bool seen[N];
        memset(seen, 0, sizeof (seen));
        if (bpm(bpGraph, u, seen, matchR))
            result++;
    }
    return result;
}

int main() {

    bool bpGraph[M][N] = {
        {0, 1, 1, 0, 0, 0},
        {1, 0, 0, 1, 0, 0},
        {0, 0, 1, 0, 0, 0},
        {0, 0, 1, 1, 0, 0},
        {0, 0, 0, 0, 0, 0},
        {0, 0, 0, 0, 0, 1}};

    cout << "Maximum number of applicants that can get job is "
            << maxBPM(bpGraph);

    return 0;
} 

File: Recien traducido/Grafos/MaximumBipartiteMatching.py

class MBM: 
    def __init__(self, graph): 
        self.graph = graph  
        self.ppl = len(graph) 
        self.jobs = len(graph[0]) 
    def bpm(self, u, matchR, seen): 
        for v in range(self.jobs): 
            if self.graph[u][v] and seen[v] == False: 
                seen[v] = True 
                if matchR[v] == -1 or self.bpm(matchR[v],matchR, seen): 
                    matchR[v] = u 
                    return True
        return False

    def maxBPM(self): 
        matchR = [-1] * self.jobs 
        result = 0 
        for i in range(self.ppl): 
            seen = [False] * self.jobs 
            if self.bpm(i, matchR, seen): 
                result += 1
        return result 
  
  
bpGraph = [[0, 1, 1, 0, 0, 0], 
    [1, 0, 0, 1, 0, 0], 
    [0, 0, 1, 0, 0, 0], 
    [0, 0, 1, 1, 0, 0], 
    [0, 0, 0, 0, 0, 0], 
    [0, 0, 0, 0, 0, 1]] 
  
g = MBM(bpGraph) 
  
print ("Maximum number of applicants that can get job is %d " % g.maxBPM()) 
  